CONTRIBUTING.md

```diff
@@ -627,6 +631,74 @@ required.
 Essentially, the steps you have to take to deploy manually are the same that
 `skaffold` will perform, and they're outlined at the previous section.
 
+#### Kubernetes E2E tests
+
+Kubernetes integration is quite complicated, a lot of things can go wrong.
+
+To cope with the complexity and ensure we maintain high quality, we use
+E2E (end-to-end) tests.
+
+> E2E tests normally run at CI, so there's typically no need to run them
+> manually.
+
+##### Requirements
+
+* `kubernetes` cluster (`minikube` has special support, but any cluster should
+  work)
+* `docker`
+* `kubectl`
+* `bash`
+
+Vector release artifacts are prepared for E2E tests, so the ability to do that
+is required too, see Vector [docs](https://vector.dev) for more details.
+
+##### Running the E2E tests
+
+To run the E2E tests, use the following command:
+
+```shell
+CONTAINER_IMAGE_REPO=<your name>/vector-test make test-e2e-kubernetes
+```
+
+Where `CONTAINER_IMAGE_REPO` is the docker image repo name to use, without part
+after the `:`. Replace `<your name>` with your Docker Hub username.
+
+This will run Kubernetes E2E tests suite. If the command exit with 0 status
+code - tests passed.
+
+You can also pass additional parameters to adjust the behavior of the test:
```

Comment:
I suggest you namespace the k8s specific variables to like `KUBERNETES_USE_CACHE` and `KUBERNETES_CONTAINER_IMAGE_REPO` so we don't pollute the global environment namespace with variables whose names are relatively unclear.

Reply:
Prefixing all of them with `KUBERNETES_` is way too much characters to type, but `K8S_` would be fine I guess.
That said, I don't think global env pollution is an issue here. Unless the variables set globally at the system level (i.e. like in `~/.profile`) - but then it'd require more than just `K8S_` for them to make sense universally. I woundn't worry about that case.

Given that those are used in a very specific command, I don't think it's big issue. It's easy to type and remember though - and this is a big plus IMO.
If using them in `.envrc` or similar - it's often possible to use comments.